terraform/files/template/cluster-template.yaml

```diff
@@ -63,15 +63,64 @@ spec:
       controllerManager:
         extraArgs:
           cloud-provider: external
+      etcd:
+        local:
+          dataDir: /var/lib/etcd
+          extraArgs:
+            heartbeat-interval: "250"
+            election-timeout: "2500"
+            snapshot-count: "6400"
+            auto-compaction-mode: periodic
+            auto-compaction-retention: 8h
     joinConfiguration:
       nodeRegistration:
         name: '{{ local_hostname }}'
         kubeletExtraArgs:
           cloud-provider: external
+    files:
+      - path: /etc/systemd/system/etcd-defrag.service
+        owner: "root:root"
+        permissions: "0644"
+        content: |
+          [Unit]
+          Description=Run etcdctl defrag
+          Documentation=https://etcd.io/docs/v3.3.12/op-guide/maintenance/#defragmentation
```

Comment:
we should update this to the current used version (latest: https://github.com/etcd-io/etcd/releases)